### PR TITLE
made resource locator reload on every rlp.part change

### LINKED ISSUE #1984: Multiple `sulu.rlp.part` tags in template cause never ending loader
When multiple `sulu.rlp.part` are defined you have to fill in both before saving. because otherwise you get no error but an never ending loader on the save button.

example: 

```
<section name="highlight">
    <properties>
        <property name="title" type="text_line" mandatory="true">
            <meta>
                <title lang="de">Titel</title>
                <title lang="en">Title</title>
            </meta>
            <params>
                <param name="headline" value="true"/>
            </params>
​
            <tag name="sulu.rlp.part"/>
        </property>
​
        <property name="sub_title" type="text_line">
            <meta>
                <title lang="de">Subtitle</title>
                <title lang="en">Subtitle</title>
            </meta>
​
            <tag name="sulu.rlp.part"/>
        </property>
​
        <property name="url" type="resource_locator" mandatory="true">
            <meta>
                <title lang="de">Adresse</title>
                <title lang="en">Resourcelocator</title>
            </meta>
​
            <tag name="sulu.rlp"/>
        </property>
    </properties>
</section>
````

maybe we should couple the tag to mandatory properties.

### PR DESCRIPTION
**Pull request by marcelmoosbrugger**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #1984
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

In the content section, when creating a new page, the resource locator now gets always gets newly generated if a resource-locator part changes.

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/form/main.js [hunk 1]
```diff
@@ -30,17 +30,15 @@ define(['app-config', 'config', 'services/sulupreview/preview'], function(AppCon
 
         template: '',
 
+        add: true,
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/form/main.js]: `add` sounds like a function

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/form/main.js [hunk 2]
```diff
@@ -60,19 +58,31 @@ define(['app-config', 'config', 'services/sulupreview/preview'], function(AppCon
...
-            this.getDomElementsForTagName('sulu.rlp', function(property) {
-                var element = property.$el.data('element');
-                if (!element || element.getValue() === '' || element.getValue() === undefined || element.getValue() === null) {
-                    this.startListening = true;
-                }
-            }.bind(this));
+            if (!!this.add) {
+                // when a rlp-part gets changed other parts (e.g. submitting) has to wait
+                this.sandbox.dom.on(this.getDomElementsForTagName('sulu.rlp.part'), 'keyup', function() {
+                    this.whenResourceLocatorIsLoaded = $.Deferred();
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/form/main.js]: This object is generated for every `keyup`? Shouldn't we only do that if the promise haven't been resolved yet?

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/form/main.js [hunk 3]
```diff
@@ -60,19 +58,31 @@ define(['app-config', 'config', 'services/sulupreview/preview'], function(AppCon
...
+                }.bind(this));
+
+                this.sandbox.dom.on(this.getDomElementsForTagName('sulu.rlp.part'), 'focusout', this.loadResourceLocator.bind(this));
+
+                // initialize all rlp-inputs as not edited by the user
+                this.getDomElementsForTagName('sulu.rlp', function(property) {
+                    property.$el.data('user-edited', false);
+                    // manually edited rlp-inputs get marked as such
+                    this.sandbox.dom.on(property.$el, 'change', function() {
+                        property.$el.data('user-edited', property.$el.data('element').getValue().length > 1);
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/form/main.js]: So if a part is completely deleted it is not recognized as edited?

Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/form/main.js]: Not in that sense that its included in the generation process of the resource locator. Only the non empty rlp fields get sent to the server.

**Comment by marcelmoosbrugger** [comment]
@danrot. Your comments are fixed, if you want to include this PR in the release :)

**Comment by danrot** [comment]
Tests are failing

**Pull request merged**